src/app/InteractionModelEngine.h

```diff
@@ -306,6 +306,9 @@ class InteractionModelEngine : public Messaging::UnsolicitedMessageHandler,
 
     CHIP_ERROR ResumeSubscriptions();
 
+    // Return the state of a subscription for a given subject (CAT or NodeId)
+    bool IsSubjectSubscriptionActive(const FabricIndex & aFabricIndex, const NodeId & subject);
```

Comment:
Depending on the answer to the "what about the case when we are in the middle of setting up that subscription", this API's name might need to change.

Reply:
So this should be called `SubjectHasActiveSubscription` or something.

Also, why is aFabricIndex not passed by value?  It's an 8-bit integer....  Passing the NodeId by reference maybe makes sense (or not), depending on calling conventions for 64-bit ints, but the 8-bit thing should just be passed by value.